src/main/ruby/core/truffle/interop.rb

```diff
@@ -36,6 +36,17 @@ def self.ruby_object_keys(object)
         end
       end
     end
+    
+    def self.lookup_symbol(name)
+      # TODO CS 9-Aug-17 I can't see to check if the method is defined
+      if method = Object.method(name) rescue nil
+        method
+      elsif Truffle::SymbolOperations.is_constant?(name) && Object.const_defined?(name)
+        Object.const_get(name)
+      else
```

Comment:
What's the difference? You need to be careful to capture the top-level self, not `self` just outside this method which == `Truffle::Interop`. You might also probably want `TOPLEVEL_SELF.respond_to?(name, true)` so it picks up methods defined with just `def` at the top-level (which are private instance methods of Object).